comps/cores/mega/gateway.py

```diff
@@ -985,18 +1044,25 @@ async def handle_request(self, request: Request):
         if "text" in result_dict[last_node].keys():
             response = result_dict[last_node]["text"]
         else:
-            # text in not response message
+            # text is not in response message
             # something wrong, for example due to empty retrieval results
             if "detail" in result_dict[last_node].keys():
                 response = result_dict[last_node]["detail"]
             else:
-                response = "The server fail to generate answer to your query!"
+                response = "The server failed to generate an answer to your query!"
         if "metadata" in result_dict[last_node].keys():
             # from retrieval results
-            metadata = result_dict[last_node]["metadata"]
+            if decoded_audio_input:
+                metadata = {"audio": decoded_audio_input}
```

Comment:
The decoded audio should be added to the existing metadata, not replace it. I am not seeing the retrieved image data in the response, so please change this to: `metadata["audio"] = decoded_audio_input`